### PR TITLE
[AutoMM] Support model parameter number statistics

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*
Support calling `predictor.total_parameters `, `predictor.trainable_parameters `, and `predictor.model_size `.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/predictor.py [hunk 1]
```diff
@@ -508,6 +509,26 @@ def column_types(self):
...
+    def trainable_parameters(self) -> int:
+        return sum(
+            p.numel() if not is_lazy_weight_tensor(p) else 0 for p in self._model.parameters() if p.requires_grad
+        )
+
+    @property
+    def model_size(self) -> float:
+        precision_to_bits = {64: 64, 32: 32, 16: 16, "bf16": 16}
+        if self._config is not None:
+            precision = precision_to_bits.get(self._config.env.precision, 16)
```
Comment (tonyhoo) [inline multimodal/src/autogluon/multimodal/predictor.py:526]: wondering how mixed precision can be handled here? Shall we use `total_size_in_bytes = sum(p.numel() * p.element_size() for p in  self._model.parameters())` to address it?

File: multimodal/src/autogluon/multimodal/utils/model.py [hunk 2]
```diff
@@ -556,3 +557,15 @@ def modify_duplicate_model_names(
 
 def list_timm_models(pretrained=True):
     return timm.list_models(pretrained=pretrained)
+
+
+def is_lazy_weight_tensor(p: Tensor) -> bool:
+    from torch.nn.parameter import UninitializedParameter
+
+    if isinstance(p, UninitializedParameter):
+        warnings.warn(
```
Comment (tonyhoo) [inline multimodal/src/autogluon/multimodal/utils/model.py:566]: Wondering how frequent this gonna happen? If it is the case, shall we just run one dummy forward pass to make these layers initialized and then calculate?

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/utils/model.py:566]: This refers to pytorch lightning: https://github.com/Lightning-AI/lightning/blob/984f49f7195ddc67e961c7c498ee6e19fc0cecb5/src/lightning/pytorch/utilities/model_summary/model_summary.py#L236
In some cases, parameters are uninitialized.

Comment (tonyhoo) [reply multimodal/src/autogluon/multimodal/utils/model.py:566]: ok, as long as we are consistent with lightning, shall be good

File: multimodal/src/autogluon/multimodal/predictor.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/predictor.py]: Good suggestion. Updated.

**Review by FANGAreNotGnu — APPROVED** [review]
LGTM

**Review by suzhoum — APPROVED** [review]
LGTM!

**Pull request merged**